File: src/main/java/org/crafter/engine/delta/DeltaObject.java
package org.crafter.engine.delta;

/**
 * Useful for testing thread performance!
 */
public class DeltaObject {
    private float deltaTime = 0;

    private long oldTime;

    public DeltaObject(){
        oldTime = System.nanoTime();
    }

    public void calculateDelta() {
        long time = System.nanoTime();
        deltaTime = (float)(time - oldTime) / 1_000_000_000.0f;
        oldTime = time;
    }

    public float getDelta() {
        return deltaTime;
    }
}

File: src/main/kotlin/org/crafter/engine/delta/DeltaObject.kt
package org.crafter.engine.delta

/**
 * Useful for testing thread performance!
 */
class DeltaObject {
    var delta = 0f
        private set
    private var oldTime: Long

    init {
        oldTime = System.nanoTime()
    }

    fun calculateDelta() {
        val time = System.nanoTime()
        delta = (time - oldTime).toFloat() / 1000000000.0f
        oldTime = time
    }
}
